app/models/order.rb

```diff
@@ -263,6 +265,35 @@ def close_direct!(user)
     update_attributes! state: 'closed', updated_by: user
   end
 
+  def send_to_supplier(user)
+    Mailer.order_result_supplier(user, self).deliver_now
+    update!(last_sent_mail: Time.now)
+  end
+
+  def do_end_action
+    if auto_close?
+      finish! created_by
+    elsif auto_close_and_send?
+      finish! created_by
+      send_to_supplier created_by
+    elsif auto_close_and_send_min_quantity?
+      finish! created_by
+      send_to_supplier created_by if order.sum >= order.supplier.min_order_quantity
+    end
```

Comment:
In many other places we use `finish!(user)`, maybe here as well? (Also `send_to_supplier`.)